rero_ils/query.py

```diff
@@ -49,28 +49,48 @@ def and_search_factory(self, search, query_parser=None):
     :param search: Elastic search DSL search instance.
     :returns: Tuple with search instance and URL arguments.
     """
-    def _default_parser(qstr=None):
+    def _default_parser(qstr=None, query_boosting=[]):
         """Default parser that uses the Q() from elasticsearch_dsl."""
         if qstr:
-            return Q('query_string', query=qstr, default_operator='AND')
+            if len(query_boosting) == 0 or ':' in qstr:
+                return Q('query_string', query=qstr, default_operator='AND')
+            if qstr and len(query_boosting) > 0:
+                return Q('bool', should=[
```

Comment:
We need to find an other approach as boosting does not use query string.

Reply:
done